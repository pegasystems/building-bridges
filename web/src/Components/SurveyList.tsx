import React from 'react';
import SurveyLink from './SurveyLink';
import * as Models from '../Models'

interface SurveyListProps {
    surveys: Models.Survey[];
}

interface SurveyListState {
    initSurveyKeys: string[]
}

export default class SurveyList extends React.Component<SurveyListProps, SurveyListState> {

    state = {
        initSurveyKeys: [] as string[]
    }

    componentDidMount(): void {
        this.setState((state, props) => ({initSurveyKeys:  props.surveys.map(survey => survey.key)}));
    }

    render(): JSX.Element {
        const surveyElements = this.props.surveys.map(survey => {
            const surveyUrl = `surveys/${survey.key}`;
            return (
                <tr className={this.state.initSurveyKeys.includes(survey.key) ? '' : 'new'} key={survey.key}>
                    <td>
                        <a href={surveyUrl}>{survey.title}</a>
                    </td>
                    <td className="fixed aligned-center">
                        {survey.viewsNumber}
                    </td>
                    <td className="fixed aligned-center">
                        {survey.questionersNumber}
                    </td>
                    <td className="fixed aligned-center">
                        {survey.votersNumber}
                    </td>
                    <td className="fixed aligned-center">
                        {new Date(survey.date).toLocaleString()}
                    </td>
                    <td className="fixed">
                        <SurveyLink link={`${window.location.href}${surveyUrl}`}/>
                    </td>
                    <td className="fixed">
                        <a href={`${surveyUrl}?results_secret=${survey.results_secret}`}>Show results</a>
                    </td>
                    <td className="fixed">
                        <a href={`${surveyUrl}?admin_secret=${survey.admin_secret}`}>Admin (Don't share this link)</a>
                    </td>
                </tr>
            );
        });

        if (surveyElements.length) {
            return (
                <div className="survey-list">
                    <table className="table survey-list">
                        <thead>
                        <tr>
                            <th colSpan={1}>
                                <h4>Your surveys</h4>
                            </th>
                            <th colSpan={1} className="aligned-center">
                                <h4>Unique visitors</h4>
                            </th>
                            <th colSpan={1} className="aligned-center">
                                <h4>Question authors</h4>
                            </th>
                            <th colSpan={1} className="aligned-center">
                                <h4>Voters</h4>
                            </th>
                            <th colSpan={1} className="aligned-center">
                                <h4>Created</h4>
                            </th>
                            <th colSpan={3}>
                            </th>
                        </tr>
                        </thead>
                        <tbody>
                        {surveyElements}
                        </tbody>
                    </table>
                </div>
            );
        }
        return <></>;
    }
}
