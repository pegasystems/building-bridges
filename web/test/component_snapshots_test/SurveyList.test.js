import SurveyList from '../../src/Components/SurveyList';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';


test('SurveyList renders properly test', () => {
    const component = renderer.create(
        <SurveyList surveys={[{
            date: '2020-01-26T13:51:50.417Z',
            hideVotes: true,
            key: 'title-1',
            results_secret: 'cf9ff3424c90759a',
            admin_secret: 'cf9ff3424c90759b',
            title: 'title'
        }]}
        />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
