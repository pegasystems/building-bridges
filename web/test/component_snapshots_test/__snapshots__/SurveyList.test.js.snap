// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SurveyList renders properly test 1`] = `
<div
  className="survey-list"
>
  <table
    className="table survey-list"
  >
    <thead>
      <tr>
        <th
          colSpan={1}
        >
          <h4>
            Your surveys
          </h4>
        </th>
        <th
          className="aligned-center"
          colSpan={1}
        >
          <h4>
            Unique visitors
          </h4>
        </th>
        <th
          className="aligned-center"
          colSpan={1}
        >
          <h4>
            Question authors
          </h4>
        </th>
        <th
          className="aligned-center"
          colSpan={1}
        >
          <h4>
            Voters
          </h4>
        </th>
        <th
          className="aligned-center"
          colSpan={1}
        >
          <h4>
            Created
          </h4>
        </th>
        <th
          colSpan={3}
        />
      </tr>
    </thead>
    <tbody>
      <tr
        className=""
      >
        <td>
          <a
            href="surveys/title-1"
          >
            title
          </a>
        </td>
        <td
          className="fixed aligned-center"
        />
        <td
          className="fixed aligned-center"
        />
        <td
          className="fixed aligned-center"
        />
        <td
          className="fixed aligned-center"
        >
          1/26/2020, 1:51:50 PM
        </td>
        <td
          className="fixed"
        >
          <i
            className="copy"
            onClick={[Function]}
          >
            Copy link
          </i>
        </td>
        <td
          className="fixed"
        >
          <a
            href="surveys/title-1?results_secret=cf9ff3424c90759a"
          >
            Show results
          </a>
        </td>
        <td
          className="fixed"
        >
          <a
            href="surveys/title-1?admin_secret=cf9ff3424c90759b"
          >
            Admin (Don't share this link)
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
